### ISSUE TITLE
Dependency resolution fails in Docker

### ISSUE THREAD

**Issue by gaba-xyz** [sbt/sbt#3466]
I have a few dependencies that are hosted on an on-premise Nexus server, a resolver for these dependencies has been added and there is no issue building the project. However when trying to build the project using the `openjdk:8-jre` image module resolution is failing for those dependencies.

Furthermore all dependency resolution is incredibly slow, taking easily 5-6x the normal time of resolution.

sbt version: 0.13.15

**Comment by themerius** [comment]
I have the same problem: `sbt` is very slow in dependency resolution when executed within a docker container. Any ideas?

**Comment by devshorts** [comment]
I also have this, with jdk9 and sbt launcher 1.1 and project sbt 0.13.17

**Comment by redbaron** [comment]
Are you sure it is not because your ~/.ivy2 is not available when running in docker?

**Comment by devshorts** [comment]
Turned out for me that the issue was a mounted `~/.ivy2` folder to my local host on osx.  The file watch bindings in docker on osx are abysmally slow (https://forums.docker.com/t/file-access-in-mounted-volumes-extremely-slow-cpu-bound/8076). If I instead _didn't_ mount the local ivy folder, things were much faster.  I experimented with some other tools like https://github.com/EugenMayer/docker-sync/wiki/docker-sync-on-OSX but I gave up on the process since it was all super hacky.  Figured I'd just chime in here.
